addons/l10n_ch/models/account_invoice.py

```diff
@@ -79,16 +103,20 @@ def _space_isr_number(isr_number):
             return res
 
         for record in self:
-            if record.name and record.invoice_partner_bank_id and record.invoice_partner_bank_id.l10n_ch_postal:
-                invoice_issuer_ref = re.sub('^0*', '', record.invoice_partner_bank_id.l10n_ch_postal)
+            isr_number = False
+            isr_number_spaced = False
+            if record.name and record.l10n_ch_isr_subscription:
+                invoice_issuer_ref = re.sub('^0*', '', record._get_isrb_id_number())
                 invoice_issuer_ref = invoice_issuer_ref.ljust(l10n_ch_ISR_NUMBER_ISSUER_LENGTH, '0')
                 invoice_ref = re.sub('[^\d]', '', record.name)
                 #We only keep the last digits of the sequence number if it is too long
                 invoice_ref = invoice_ref[-l10n_ch_ISR_NUMBER_ISSUER_LENGTH:]
                 internal_ref = invoice_ref.zfill(l10n_ch_ISR_NUMBER_LENGTH - l10n_ch_ISR_NUMBER_ISSUER_LENGTH - 1) # -1 for mod10r check character
 
-                record.l10n_ch_isr_number = mod10r(invoice_issuer_ref + internal_ref)
-                record.l10n_ch_isr_number_spaced = _space_isr_number(record.l10n_ch_isr_number)
+                isr_number = mod10r(invoice_issuer_ref + internal_ref)
+                isr_number_spaced = _space_isr_number(record._get_isrb_id_number)
```

Comment:
Seems there is an error for displayable isr_number_spaced

```suggestion
                isr_number_spaced = _space_isr_number(isr_number)
```